assets/js/translator.js

```diff
@@ -661,6 +699,16 @@ function populateTranslationList() {
                 return 1;
             }
         });
+
+        for(i = 0; i < variantsOnly.length; i++) {
```

Comment:
Just to confirm - this works when a variant is possible but the parent language isn't and vice versa, right? Could you provide screenshots for each case? Also, this nested for loop seems a bit expensive?

Reply:
Yes, that's correct—I've attached a screenshot of a case when the parent language is available but the variant language isn't.  As for the converse scenario, apertium.org doesn't have a pair where the variant language is available but the parent language isn't.  However, it should work as well, as the 'for' loop inserts each variant language immediately after its respective parent language, no matter what the position is—this can be seen in the screenshot.

I agree, the 'for' loop is a bit inefficient; however, I couldn't find a better way to sort the list of languages so that each variant appears immediately after its parent language.  I'd be happy to implement another solution if we can find one. :)